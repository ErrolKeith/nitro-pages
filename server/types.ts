export interface Globals {
  siteId: string;
  companyName: string;
  phoneNumber: string;
  services: string[];
}
