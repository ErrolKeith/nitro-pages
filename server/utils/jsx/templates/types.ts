const templates = ["home", "page", "internal", "404", "index", "post"] as const;
type TemplateType = (typeof templates)[number];

export interface TemplateConfig {
  type: TemplateType;
  siteId: string;
  companyName: string;
  seoTitle: string;
  content: PageContentSection[];
}
