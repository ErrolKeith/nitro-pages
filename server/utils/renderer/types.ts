import { SiteGlobals } from "~/types";
import { TemplateType } from "../jsx/template";

export interface PageContext {
  template: TemplateType;
  slug: string;
  title: string;
  categories: string[];
  content: PageContentSection[];
}

export interface BaseRenderConfig {
  template: TemplateType;
  clientGlobals: SiteGlobals;
  pageContext: PageContext;
}

export interface PageRenderConfig extends BaseRenderConfig {
  template: "page" | "internal" | "post" | "index";
}

export type HomeRenderConfig = Omit<BaseRenderConfig, "template">;
