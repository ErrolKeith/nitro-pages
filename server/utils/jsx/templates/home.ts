import { TemplateConfig } from "./types";
import { makeHtml } from "../tags/html";
import { makeHead } from "../tags/head";
import { makeMain } from "../tags/main";
import { makeTitle } from "../tags/title";

export function makeHomeTemplate(config: TemplateConfig) {
  const { siteId, seoTitle, content } = config;

  const pageContent = content.map((content) => {
    return makeParagraph({ text: "home" });
  });

  return makeHtml({
    props: {
      ["data-site-id"]: siteId,
    },
    children: [
      makeHead({ props: {}, children: [makeTitle({ title: seoTitle })] }),
      makeMain({ props: {}, children: pageContent }),
    ],
  });
}
