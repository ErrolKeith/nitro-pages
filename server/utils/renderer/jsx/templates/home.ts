import { TemplateConfig } from "./types";

export function makeHomeTemplate(config: TemplateConfig) {
  const { siteId, seoTitle, content } = config;

  const pageContent = content.map((content) => {
    return makeParagraph({ text: content.type });
  });

  return makeHtml({
    props: {
      ["data-site-id"]: siteId,
    },
    children: [
      makeHead({ props: {}, children: [makeTitle({ title: seoTitle })] }),
      makeMain({ props: {}, children: pageContent }),
    ],
  });
}
