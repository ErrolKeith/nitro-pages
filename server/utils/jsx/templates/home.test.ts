import { it, expect } from "vitest";
import { makeHomeTemplate } from "./home";
import { TemplateConfig } from "./types";

it("can make a home page", () => {
  const testTemplateConfig: TemplateConfig = {
    type: "home",
    siteId: "124246796763",
    companyName: "sample company name",
    seoTitle: "sample home",
    content: [],
  };
  const templateHomeTest = makeHomeTemplate(testTemplateConfig);
  expect(templateHomeTest).toMatchSnapshot();
});
