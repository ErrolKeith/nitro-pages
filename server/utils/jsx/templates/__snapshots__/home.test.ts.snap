// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make a home page 1`] = `
<html
  data-site-id="124246796763"
>
  <head>
    <title>
      sample home
    </title>
  </head>
  <main />
</html>
`;
